### ISSUE TITLE
Reproducing the Fold Classification baselines

### ISSUE THREAD

**Issue by anonymous-0545** [a-r-j/ProteinWorkshop#99]
Hello, thank you for the great and detailed work. I was trying to reproduce the results for fold classification with one model (GCPNet), however, after a few runs it was apparent that it doesn't align with the results presented in the paper. Here is the command used:

```
# reproduce the baseline tasks sweep (i.e., those performed without pre-training each model)
wandb sweep proteinworkshop/config/sweeps/baseline_fold.yaml
```

I slightly modified the `baseline_fold.yaml` to report the results on wandb and only use the GCPNet model. Here is the updated file:

```
program: proteinworkshop/train.py
method: grid
name: baseline_gc_prediction
metric: # Does not matter, as we are using sweep to run the experiment.
  goal: minimize
  name: val/loss/total

parameters:
  task:
    values: [multiple_graph_classification]

  dataset:
    values: [fold_family, fold_superfamily, fold_fold]

  encoder:
    values: [gcpnet]

  optimiser.optimizer.lr:
    value: 0.0001

  features:
    values: [ ca_seq,ca_bb]

  scheduler:
    values: [plateau]

  extras.enforce_tags:
    value: False

  +aux_task:
    values: [none]

  trainer:
    value: gpu

  trainer.max_epochs:
    value: 150

  test:
    value: True


  name:
    value: "${hydra:runtime.choices.encoder}_${hydra:runtime.choices.features}_${hydra:runtime.choices.dataset}"

  logger:
    value: wandb

command:
  - ${env}
  - HYDRA_FULL_ERROR=1
  - WANDB_START_METHOD=thread
  - python
  - ${program}
  - ${args_no_hyphens}
```

You can find the results I'm getting below (all 99.99+). 
![image](https://github.com/user-attachments/assets/ec8743ad-37fe-4695-9cad-6e35845d73a9)

Any suggestions how to correct this issue?

**Comment by a-r-j** [comment]
Hi @anonymous-0545 

What do the training and validation curves look like?

One possible thing to try is switching the task:

```
parameters:
  task:
    values: [multiclass_graph_classification]
```

I don't exactly remember which task name is correct but I'll dig into the logs to confirm. Please do try this and let me know how you get on though.

 One more thing to note is that it doesn't look like you're using the hparams (e.g. lr and dropout). See the separate hparams config for those values.

**Comment by anonymous-0545** [comment]
Thank you for the prompt response. The curves start around 99.9, as you can see below:
![image](https://github.com/user-attachments/assets/06774fb1-2b3d-4672-99ba-867062bbe5f7)

I've added the lr and dropout to the config as follows:
```
  optimiser.optimizer.lr:
    value: ${hparams.hparams.lr}

  decoder.graph_label.dropout:
    value: ${hparams.hparams.decoder_dropout}
```

I've started with `multiclass_graph_classification` but it would be great if you could confirm which one is used on the experiments. The initial validation accuracies are 0.07, 0.13, 0.18. Also, why is separate training needed for different test sets? Is there a difference in the training or validation set for each?

**Comment by a-r-j** [comment]
> I've started with multiclass_graph_classification but it would be great if you could confirm which one is used on the experiments.

I checked and it should be `multiclass_graph_classification`

> The initial validation accuracies are 0.07, 0.13, 0.18

That looks like it lines up with my logs :)

> Also, why is separate training needed for different test sets? Is there a difference in the training or validation set for each?

No should not be necessary, we initially implemented them separately but since removed the need for separate training.

**Comment by a-r-j** [comment]
@anonymous-0545 How did you get on? Can I close the issue?

**Comment by a-r-j** [comment]
Closing. Feel free to re-open if not fully addressed.
